python/ml4ir/applications/classification/tests/test_base.py

```diff
@@ -72,7 +75,29 @@ def setUp(
         # Setup logging
         outfile: str = os.path.join(self.args.logs_dir, "output_log.csv")
 
-        self.logger = setup_logging(reset=True, file_name=outfile, log_to_file=True)
+        self.logger = setup_logging(reset=True,
+                                    file_name=outfile,
+                                    log_to_file=True)
+        self.run_default_pipeline(data_format="csv")
```

Comment:
I preferred to run the training pipeline as a part of `setUp`. This way, it is run only once and then in the `test_classification_model.py` we can just evaluate the training results.

Reply:
Too late to the party, but `setUp` and `tearDown` run before and after **each test** in the class. We need to change this to [setUpClass](https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUpClass) for it to run only once. cc: @lastmansleeping @jakemannix 

I'll try this locally to check if it helps with the memory issue and create a PR